src/main/scala/firrtl/transforms/EnsureNamedStatements.scala

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package firrtl.transforms
+
+import firrtl._
+import firrtl.ir._
+
+/** Adds default names to print, stop and verification statements if their name is empty. */
+object EnsureNamedStatements extends Transform with DependencyAPIMigration {
+  override def invalidates(a: Transform) = false
+
+  override protected def execute(state: CircuitState): CircuitState = {
+    val c = state.circuit.mapModule(onModule)
+    state.copy(circuit = c)
+  }
+
+  private def onModule(m: DefModule): DefModule = m match {
+    case e:   ExtModule => e
+    case mod: Module =>
+      val namespace = Namespace(mod)
+      mod.mapStmt(onStmt(namespace))
+  }
+
+  private def onStmt(namespace: Namespace)(stmt: Statement): Statement = stmt match {
+    case s: Print if s.name.isEmpty => s.withName(makeName(namespace, "print"))
+    case s: Stop if s.name.isEmpty => s.withName(makeName(namespace, "stop"))
+    case s: Verification if s.name.isEmpty =>
+      val baseName = s.op match {
+        case Formal.Cover  => "cover"
+        case Formal.Assert => "assert"
+        case Formal.Assume => "assume"
+      }
+      s.withName(makeName(namespace, baseName))
+    case other => other.mapStmt(onStmt(namespace))
+  }
+
+  // We are using a slightly different algorithm than the basic namespace.
+  // We want all generated names to end in a number like `assert_0`, `assert_1`, etc.
+  private def makeName(namespace: Namespace, base: String): String = {
+    val name = Iterator.from(0).map(i => s"${base}_$i").find(n => !namespace.contains(n)).get
+    namespace.newName(name)
+  }
```

Comment:
```suggestion
```
Instead of all of this logic, you could just seed the `Namespace` with the base names to ensure it always starts with `_0`, that's what the [namespace itself does for `_GEN`](https://github.com/chipsalliance/firrtl/blob/6e0e760526090c694ce6507db71122654ffc3000/src/main/scala/firrtl/Namespace.scala#L11).

Reply:
I will change to your suggested solution.